### PR TITLE
mail-filter/mimedefang:

### PR DESCRIPTION
**Pull request by vieridipaola**
version bump and a few fixes

Bug: https://bugs.gentoo.org/866662
Bug: https://bugs.gentoo.org/801646
Tested-by: Vieri Di Paola <rentorbuy@yahoo.com>
Signed-off-by: Vieri Di Paola <rentorbuy@yahoo.com>
Package-Manager: Portage-3.0.30, Repoman-3.0.3

### CODE REVIEW

**Comment by thesamesam** [comment]
No need for Tested-by for your own commits. Also, please squash the two commits.

File: mail-filter/mimedefang/mimedefang-3.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (juippis) [inline mail-filter/mimedefang/mimedefang-3.1.ebuild:4]: 8 please, looks like it's going to work.

File: mail-filter/mimedefang/mimedefang-3.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+DESCRIPTION="Antispam, antivirus and other customizable filters for MTAs with Milter support"
+HOMEPAGE="http://www.mimedefang.org/"
```
Comment (juippis) [inline mail-filter/mimedefang/mimedefang-3.1.ebuild:7]: `https://`

File: mail-filter/mimedefang/mimedefang-3.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+RDEPEND="
+	acct-group/defang
+	acct-user/defang
+	dev-perl/Digest-SHA1
+	dev-perl/IO-stringy
+	dev-perl/MailTools
+	dev-perl/MIME-tools
+	dev-perl/Unix-Syslog
+	|| ( mail-mta/sendmail mail-mta/postfix )
```
Comment (juippis) [inline mail-filter/mimedefang/mimedefang-3.1.ebuild:24]: Could you move this to be the first, or the last dep before conditions?

File: mail-filter/mimedefang/mimedefang-3.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		dev-perl/Test-Most
+	)
+"
+
+src_prepare() {
+	eapply "${FILESDIR}/${PN}-2.72-ldflags.patch"
+	eapply "${FILESDIR}/${PN}-vendor.patch"
+	eapply "${FILESDIR}/${PN}-tests2.patch"
+	eapply_user
+}
```
Comment (juippis) [inline mail-filter/mimedefang/mimedefang-3.1.ebuild:43]: if you use a `PATCHES=( )` array you don't have to define `src_prepare` at all here.

**Review by juippis — COMMENTED** [review]
I was going to fix the points I reviewed on my own, but I'm having troubles emerging some deps so while I figure those out something for you to do :)

**Comment by vieridipaola** [comment]
2e692a3aff13dd77351c4546b21f08466366f62a has changed files/mimedefang-tests2.patch as suggested upstream

**Comment by arkamar** [comment]
This is a bit obsolete now, version 3.4.1 is available upstream.

**Comment by eli-schwartz** [comment]
This was actually merged but without a Closes tag! bb2427ffa72ef11c0589318af1f4ac51861dae95

```
commit bb2427ffa72ef11c0589318af1f4ac51861dae95
Author:     Vieri Di Paola <rentorbuy@yahoo.com>
AuthorDate: 2022-08-29 09:48:00 +0000
Commit:     Sam James <sam@gentoo.org>
CommitDate: 2022-10-25 05:51:07 +0000

    mail-filter/mimedefang: version bump to 3.1 and several fixes
    
    * PID dir according to upstream changelog (2017)
    * Perl vendor path
    * Unit test
    * Postfix and sendmail dependencies (virtual/mta too broad)
    
    Closes: https://bugs.gentoo.org/801646
    Closes: https://bugs.gentoo.org/866662
    Signed-off-by: Vieri Di Paola <rentorbuy@yahoo.com>
    Package-Manager: Portage-3.0.30, Repoman-3.0.3
    Signed-off-by: Sam James <sam@gentoo.org>

 mail-filter/mimedefang/Manifest                    |  1 +
 .../mimedefang/files/mimedefang-tests2.patch       | 22 ++++++
 .../mimedefang/files/mimedefang-vendor.patch       | 32 +++++++++
 mail-filter/mimedefang/files/mimedefang.conf       |  3 +
 mail-filter/mimedefang/files/mimedefang.init       |  9 ++-
 mail-filter/mimedefang/mimedefang-3.1.ebuild       | 79 ++++++++++++++++++++++
 6 files changed, 143 insertions(+), 3 deletions(-)
```

**Comment by thesamesam** [comment]
It's a common mistake (for me) when merging a PR that has a commit that requires squashing :(

**Comment by eli-schwartz** [comment]
That's one advantage of asking the submitter to squash, I suppose...

**Pull request closed without merging**